lib/ex_aws/rekognition.ex

```diff
@@ -2,37 +2,126 @@ defmodule ExAws.Rekognition do
+  """
+  @spec compare_faces(number(), binary() | S3Object.t(), binary() | S3Object.t()) :: %{
+          optional(any) => any
+        }
+  def compare_faces(similarity_threshold \\ 0.8, source_image, target_image)
+      when is_number(similarity_threshold) do
+    request(:compare_faces, %{
+      "SimilarityThreshold" => similarity_threshold,
+      "SourceImage" => map_image(source_image),
+      "TargetImage" => map_image(target_image)
+    })
+  end
 
-    data = %{
-      "Image" => %{
-        "Bytes" => image_64
-      }
-    }
+  @doc """
+  https://docs.aws.amazon.com/rekognition/latest/dg/API_DeleteFaces.html
+  """
+  @spec delete_faces(binary(), maybe_improper_list()) :: %{optional(any) => any}
+  def delete_faces(collection_id, face_ids) when is_binary(collection_id) and is_list(face_ids) do
+    request(:delete_collection, %{
+      "CollectionId" => collection_id,
+      "FaceIds" => face_ids
+    })
+  end
+
+  @doc """
+  https://docs.aws.amazon.com/rekognition/latest/dg/API_DetectFaces.html
 
-    request(:detect_text, data)
+  NOTE: When using an S3Object, you may need to insure that
+  the S3 uses the same region as Rekognition
+  """
+  @spec detect_faces(maybe_improper_list(), binary() | S3Object.t()) :: %{optional(any) => any}
+  def detect_faces(attributes \\ ["DEFAULT"], image) when is_list(attributes) do
+    request(:detect_faces, %{
+      "Attributes" => attributes,
+      "Image" => map_image(image)
+    })
   end
 
-  def detect_text(s3_object_bucket, s3_object_name, s3_object_version \\ nil) do
-    data = %{
-      "Image" => %{
-        "S3Object" => %{
-          "Bucket" => s3_object_bucket,
-          "Name" => s3_object_name,
-          "Version" => s3_object_version
-        }
+  @doc """
+  https://docs.aws.amazon.com/rekognition/latest/dg/API_CreateCollection.html
+  """
+  @spec create_collection(binary()) :: %{optional(any) => any}
+  def create_collection(collection_id) when is_binary(collection_id) do
+    request(:create_collection, %{
+      "CollectionId" => collection_id
+    })
+  end
+
+  @doc """
+  https://docs.aws.amazon.com/rekognition/latest/dg/API_DeleteCollection.html
+  """
+  @spec delete_collection(binary()) :: %{optional(any) => any}
+  def delete_collection(collection_id) when is_binary(collection_id) do
+    request(:delete_collection, %{
+      "CollectionId" => collection_id
+    })
+  end
+
+  @doc """
+  https://docs.aws.amazon.com/rekognition/latest/dg/API_DescribeCollection.html
+  """
+  @spec describe_collection(binary()) :: %{optional(any) => any}
+  def describe_collection(collection_id) when is_binary(collection_id) do
+    request(:delete_collection, %{
+      "CollectionId" => collection_id
+    })
+  end
+
+  @doc """
+  https://docs.aws.amazon.com/rekognition/latest/dg/API_ListCollections.html
+  """
+  @spec list_collections(integer(), binary()) :: %{optional(any) => any}
+  def list_collections(max_results, next_token)
```

Comment:
Can `max_results` be the last argument, and have a default value, like `10` or `20` ? Does this make sense ?

Reply:
In here I need some help because both max_results and next_token are optional values... It works similar to our paginator.

For this case, how do I contemplate the optionality of both arguments? Default the value to nil?